packages/devtools_app/lib/src/code_size/code_size_controller.dart

```diff
@@ -12,8 +12,12 @@ import 'package:vm_snapshot_analysis/treemap.dart';
 import '../charts/treemap.dart';
 
 class CodeSizeController {
-  ValueListenable<TreemapNode> get root => _root;
-  final _root = ValueNotifier<TreemapNode>(null);
+  /// The node set as the current root.
+  ValueListenable<TreemapNode> get currentRoot => _currentRoot;
+  final _currentRoot = ValueNotifier<TreemapNode>(null);
+
+  /// The root node of the entire tree.
+  TreemapNode topRoot;
```

Comment:
we shouldn't need to track this directly. Calling `_currentRoot.value.root` will give us the same value.

Reply:
I intended the `topRoot` to be the ultimate root that never changes once the tree is loaded in, while `currentRoot` is something that does change due to user interaction (i.e. clicking on a treemap cell). I added this in for the purpose of keep tracking the total size (which will always be `topRoot.byteSize`). Should I add clearer documentation? Or might there be a better way to handle the total size (e.g. adding a `totalSize` member for `CodeSizeController`)?

Reply:
You should be able to access the top root of the tree from any node by calling node.root (this is a getter on the TreeNode class). The actual tree data is immutable - only the UI components reference to the "root" is changing. So from any node `root`, `root.root` will return the absolute root of the tree

Reply:
Ah I did not notice that there is a root value. Got it!